05.wc/wc.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  column_names, paths = column_names_and_paths
+  statistics_matrix =
+    if paths.empty?
+      [build_statistics_row($stdin.read)]
+    else
+      build_statistics_matrix(paths)
+    end
+  width = calculate_output_width(statistics_matrix, column_names)
+  statistics_matrix.each do |statistics|
+    render(statistics, column_names, width)
+  end
+end
+
+def column_names_and_paths
+  line = false
+  word = false
+  size = false
+  opt = OptionParser.new
+  opt.on('-l') { |v| line = v }
+  opt.on('-w') { |v| word = v }
+  opt.on('-c') { |v| size = v }
+  paths = opt.parse(ARGV)
+  column_names = [line, word, size].none? ? %i[line_count word_count size] : { line_count: line, word_count: word, size: size }.select { |_, v| v }.keys
```

Comment:
ちょっと横に長すぎる気がしますね。改行した方が読みやすいかも。

```suggestion
  column_names = [line, word, size].none? ? 
    %i[line_count word_count size] : 
    { line_count: line, word_count: word, size: size }.select { |_, v| v }.keys
```